src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,645 @@
+                )?;
+
+                // Increase offset by 1 because the running sum decomposition takes
+                // up 86 rows (1 more than the number of windows.)
+                let offset = offset + 1;
+
+                // Add to the accumulator and return the final result as `[scalar]B`.
+                let result = self.super_config.add_config.assign_region(
+                    &mul_b,
+                    &acc,
+                    offset + NUM_WINDOWS,
+                    &mut region,
+                )?;
+
+                #[cfg(test)]
+                // Check that the correct multiple is obtained.
+                {
+                    use group::Curve;
+
+                    let base: super::OrchardFixedBases = base.into();
+                    let scalar = &scalar
+                        .base_field_elem()
+                        .value()
+                        .map(|scalar| pallas::Scalar::from_bytes(&scalar.to_bytes()).unwrap());
+                    let real_mul = scalar.map(|scalar| base.generator() * scalar);
+                    let result = result.point();
+
+                    if let (Some(real_mul), Some(result)) = (real_mul, result) {
+                        assert_eq!(real_mul.to_affine(), result);
+                    }
+                }
+
+                Ok((result, scalar))
+            },
+        )?;
+
+        // We want to enforce canonicity of a 255-bit base field element, α.
+        // That is, we want to check that 0 ≤ α < p, where p is Pallas base
+        // field modulus p = 2^254 + t_p
+        //                 = 2^254 + 45560315531419706090280762371685220353.
+        // Note that t_p < 2^130.
+        //
+        // α has been decomposed into three pieces,
+        //            α = α_0 (252 bits)  || α_1 (2 bits) || α_2 (1 bit).
+        //
+        // If the MSB α_2 = 1, then:
+        //      - α_2 = 1 => α_1 = 0, and
+        //      - α_2 = 1 => α_0 < t_p. To enforce this:
+        //      - α_2 = 1 => 0 ≤ α_0 < 2^130
+        //                => 13 ten-bit lookups of α_0
+        //      - α_2 = 1 => 0 ≤ α_0 + 2^130 - t_p < 2^130
+        //                => 13 ten-bit lookups of α_0 + 2^130 - t_p
+        //                => z_13_alpha_0_prime = 0
+        //
+        let (alpha, running_sum) = (scalar.base_field_elem, &scalar.running_sum);
+        let z_84_alpha = running_sum[83];
+        let z_85_alpha = running_sum[84];
+
+        let (alpha_0, z_13_alpha_0) = {
+            // α_0 = α - z_84_alpha * 2^252
+            let alpha_0 = alpha
+                .value()
+                .zip(z_84_alpha.value())
+                .map(|(alpha, z_84_alpha)| {
+                    let two_pow_252 = pallas::Base::from_u128(1 << 126).square();
+                    alpha - z_84_alpha * two_pow_252
+                });
+
+            let (alpha_0, zs) = self.lookup_config.witness_check(
+                layouter.namespace(|| "Lookup range check alpha + 2^130 - t_p"),
```

Comment:
```suggestion
                layouter.namespace(|| "Lookup range check alpha"),
```
However, we should be able to use a similar trick to https://hackmd.io/o9EzZBwxSWSi08kQ_fMIOw?view#Overflow-check-for-variable-base-scalar-multiplication to enforce α ∊ [0, 2<sup>130</sup>) without an explicit 10-bit decomposition:

Note that 132 = 44 · 3. Also we have z<sub>44</sub> = a<sub>44</sub> + 2<sup>3</sup> a<sub>45</sub> + ... + 2<sup>3·(84-44)</sup> a<sub>84</sub>.
So the low 132 bits of α are zero exactly when α - 2<sup>44</sup> z<sub>44</sub> = 0. And then we can check that α ∊ [0, 2<sup>130</sup>) by checking α - 2<sup>44</sup> z<sub>44</sub> = 0 and a<sub>43</sub> ∊ {0, 1}.

Reply:
This optimisation makes sense to me, but I understand the details as being slightly different: the range check is on α<sub>0</sub> = α - 2<sup>3·(84)</sup> a<sub>84</sub> ∊ [0, 2<sup>130</sup>). 

To do this, we check:
- the high 120 bits of α<sub>0</sub> are all-zero, i.e. z<sub>44</sub> - 2<sup>3·(84 - 44)</sup> a<sub>84</sub> = 0. This constrains α<sub>0</sub> to be a 132-bit value.
- bits 130..=131 of α<sub>0</sub>, i.e. a<sub>43</sub> ∊ {0,1}. This constrains α<sub>0</sub> to be a 130-bit value.